### DISCUSSION TITLE
[Feature request] Allow Komodo to manage its own stack

### DISCUSSION THREAD

**Discussion by Laflamme** [moghtech/komodo#223] [Ideas]
Currently, komodo cannot manage its own stack. It's possible to execute the same actions as with other stacks, but some of those actions will fail (Redeploy, Restart) and others don't really make sense for a docker management tool (Pause, Stop, Destroy). 

Maybe komodo could detect when a stack is using ghcr.io/mbecker20/komodo? It could then disable Pause, Stop and Destroy for that stack, and implement a workaround for Redeploy and Restart.

Dockge is faced with the same issue. Here is what has been proposed on their side:

> I ma also interest in this feature, You could just spawn small little container and restart it from with in ? similar way Yacht do it  () and after update again delete the container ?

_Originally posted by @GamerClassN7 in https://github.com/louislam/dockge/discussions/288#discussioncomment-7851846_

What do you think?

**Comment by mbecker20** [comment]
I always add a Stack to manage Komodo Core. I run Periphery with systemd though, so Periphery isn't part of the Komodo Core stack. In this case, Komodo _can_ redeploy itself, eg to update the version. The Update log will seem to fail due to Core restart, but Periphery is handling the redeploy, so it works despite the failed Update log. If refresh the page after a short time, you will see Core comes back up and the version has been updated.

If you prefer to run Periphery in container, you can accomplish something similar by deploying Periphery container outside of the Core stack, maybe in its own dedicated compose file, or just using docker run command.

In terms of managing Periphery version across many servers, I recommend an Ansible playbook for this. You can run Ansible in a container (`alpine/ansible` on docker hub), and deploy the Ansible container using Komodo, mounting the playbook and your host file in the container. With this setup you can have all your Periphery agents updated on button click inside Komodo, you just deploy the Ansible Stack / Deployment to run the playbook. See https://github.com/mbecker20/komodo/discussions/220.

**Reply by Laflamme** [reply]
Thank you so much! What a generous answer; I will look into these solutions for sure. Cheers!

**Comment by Solear** [comment]
I would also love if komodo could redeploy itself.

I'm to stupid to add a stack to manage komodo on its own. I' struggeling, komodo is initial installed at /opt/komodo via `docker compose -p komodo -f komodo/mongo.compose.yaml --env-file komodo/compose.env up -d` but my komodo-managed stacks are in /etc/komodo/stacks.
